File: 191-NumberOf1Bits/191-NumberOf1Bits.cpp
// Last updated: 6/7/2025, 2:42:12 AM
class Solution {
public:
     int hammingWeight(uint32_t n) {
        int count = 0;
         
        for (int i = 0; i < 32; i++) {
            count = (n >> i) & 1 ? count + 1 : count;
        }

        return count;     
    }
};

File: 191-NumberOf1Bits/191-NumberOf1Bits.rs
// Last updated: 6/7/2025, 2:42:10 AM
impl Solution {
    pub fn hamming_weight(n: i32) -> i32 
    {
        // count `1` bits using a mask (n /in [1, 2^32])
        let (mut cnt, mut mask) = (0, 1);
        for i in 0..=32
        {
            if (n & mask != 0) { cnt += 1; }
            mask <<= 1;
        }
        cnt
    }
}
